tests/qos/test_qos_dscp_mapping.py

```diff
@@ -0,0 +1,297 @@
+                                              outer_dst_mac=router_mac,
+                                              outer_src_pkt_ip=outer_src_pkt_ip,
+                                              outer_dst_pkt_ip=outer_dst_pkt_ip,
+                                              outer_dscp=outer_dscp,
+                                              inner_src_pkt_ip=inner_src_pkt_ip,
+                                              inner_dst_pkt_ip=inner_dst_pkt_ip,
+                                              inner_dscp=inner_dscp,
+                                              decap_mode=decap_mode)
+
+            queue_val = get_dscp_to_queue_value(rotating_dscp, dut_qos_maps.get("dscp_to_tc_map").get("AZURE"),
+                                                dut_qos_maps.get("tc_to_queue_map").get("AZURE"))
+
+            global output_table
+
+            if queue_val is None:
+                logger.info("No queue found for dscp {} on DUT".format(inner_dscp))
+                output_table.append([rotating_dscp, "No queue found", "N/A", "N/A"])
+                continue
+
+            # Clear queue counters
+            duthost.command("sonic-clear queuecounters")
+            time.sleep(2)
+
+            try:
+                dst_ptf_port_id = send_and_verify_traffic(ptfadapter=ptfadapter,
+                                                          pkt=pkt,
+                                                          exp_pkt=exp_pkt,
+                                                          ptf_src_port_id=ptf_src_port_id,
+                                                          ptf_dst_port_ids=ptf_dst_port_ids)
+
+            except Exception as e:
+                raise (e)
+
+            dut_egress_port = get_dut_pair_port_from_ptf_port(duthost, tbinfo, dst_ptf_port_id)
+            pytest_assert(dut_egress_port, "No egress port on DUT found for ptf port {}".format(dst_ptf_port_id))
+
+            egress_queue_count, _ = get_egress_queue_count(duthost, dut_egress_port, queue_val)
+            verification_success = abs(egress_queue_count - DEFAULT_PKT_COUNT) < TOLERANCE
+
+            if verification_success:
+                logger.info("Received expected number of packets on queue {}".format(queue_val))
+                output_table.append([rotating_dscp, queue_val, egress_queue_count, "SUCCESS"])
+            else:
+                logger.info("Received {} packets on queue {} instead of {}".format(egress_queue_count, queue_val,
+                                                                                   DEFAULT_PKT_COUNT))
+                output_table.append([rotating_dscp, queue_val, egress_queue_count, "FAILURE"])
+
+        logger.info("DSCP to queue mapping test results:\n{}"
+                    .format(tabulate(output_table,
+                                     headers=["Inner Packet DSCP Value", "Egress Queue",
+                                              "Egress Queue Count", "Result"])))
+
```

Comment:
If one of the verification failed, please fail the test case at the end. Otherwise the test will always succeed.

Reply:
Oh ok